AdvancedDLSupport/ImplementationGenerators/Wrappers/SpanMarshallingWrapper.cs

```diff
@@ -1,135 +1,240 @@
-
-        /// <inheritdoc />
-        public override bool IsApplicable(IntrospectiveMethodInfo member)
-        {
-            return member.ReturnType.IsGenericType // prevents exception on the line below
-                   && member.ReturnType.GetGenericTypeDefinition() == typeof(Span<>);
-        }
-    }
-}
+//
+//  SpanMarshallingWrapper.cs
+//
+//  Copyright (c) 2018 Firwood Software
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using AdvancedDLSupport.Extensions;
+using AdvancedDLSupport.Pipeline;
+using AdvancedDLSupport.Reflection;
+using JetBrains.Annotations;
+using StrictEmit;
+
+namespace AdvancedDLSupport.ImplementationGenerators
+{
+    /// <summary>
+    /// Generates wrapper instructions for returning <see cref="Span{T}"/> from unmanaged code
+    /// through a pointer and provided length.
+    /// </summary>
+    internal class SpanMarshallingWrapper : CallWrapperBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanMarshallingWrapper"/> class.
+        /// </summary>
+        /// <param name="targetModule">The module where the implementation should be generated.</param>
+        /// <param name="targetType">The type in which the implementation should be generated.</param>
+        /// <param name="targetTypeConstructorIL">The IL generator for the target type's constructor.</param>
+        /// <param name="options">The configuration object to use.</param>
+        public SpanMarshallingWrapper
+        (
+            [NotNull] ModuleBuilder targetModule,
+            [NotNull] TypeBuilder targetType,
+            [NotNull] ILGenerator targetTypeConstructorIL,
+            ImplementationOptions options
+        )
+            : base
+            (
+                targetModule,
+                targetType,
+                targetTypeConstructorIL,
+                options
+            )
+        {
+        }
+
+        /// <inheritdoc />
+        public override IntrospectiveMethodInfo GeneratePassthroughDefinition(PipelineWorkUnit<IntrospectiveMethodInfo> workUnit)
+        {
+            Type returnType = workUnit.Definition.ReturnType, newReturnType;

```

Comment:
Do not declare multiple variables in one line.

Reply:
Fixed.